liblangutil/SourceReferenceFormatterHuman.cpp

```diff
@@ -103,9 +105,14 @@ void SourceReferenceFormatterHuman::printSourceLocation(SourceReference const& _
 		m_stream << leftpad << ' ';
 		frameColored() << '|';
 		m_stream << ' ';
+
+		std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
```

Comment:
Is that really the right fix? Don't we have to compute the byte size in utf-8 encoding?

I think some code in UTF8.cpp can easily be adapted to compute the number of characters represented by a sequence of utf8 bytes.

Reply:
Yeah, I just noticed it's very incomplete. This only works properly if the UTF parts are before the reference location. If it's inside it fails.

Reply:
The `.size()` of a `std::string` is the byte size, of a `std::wstring` it is the amount of code points (as long as the unicode character is within 16 bytes that is)

Reply:
Actually we also can use std::u32string to get the full range. However this doesn't consider grapheme clusters but I think it's not worth fixing it to that extend.